loader/asm_offset.c

```diff
@@ -22,6 +22,10 @@
 
 // This code generates an assembly file which provides offsets to get struct members from assembly code.
 
+#if defined(__MINGW32__) && !defined(__USE_MINGW_ANSI_STDIO)
+#define __USE_MINGW_ANSI_STDIO 1
+#endif
+
```

Comment:
Add a comment explaining why this is needed.

Reply:
What is `__USE_MINGW_ANSI_STDIO`?

Reply:
For a tangible log:
```c
/code/Vulkan-Loader/loader/asm_offset.c: In function ‘main’:
/code/Vulkan-Loader/loader/asm_offset.c:121:27: warning: unknown conversion type character ‘z’ in format [-Wformat=]
  121 |             fprintf(file, "%-32s equ " SIZE_T_FMT "; %s\n", values[i].name, values[i].value, values[i].comment);
      |                           ^~~~~~~~~~~~
/code/Vulkan-Loader/loader/asm_offset.c:62:24: note: format string is defined here
   62 | #define SIZE_T_FMT "%-8zu"
      |                        ^
/code/Vulkan-Loader/loader/asm_offset.c:121:27: warning: format ‘%s’ expects argument of type ‘char *’, but argument 4 has type ‘size_t’ {aka ‘long long unsigned int’} [-Wformat=]
  121 |             fprintf(file, "%-32s equ " SIZE_T_FMT "; %s\n", values[i].name, values[i].value, values[i].comment);
      |                           ^~~~~~~~~~~~                                      ~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      size_t {aka long long unsigned int}
/code/Vulkan-Loader/loader/asm_offset.c:121:55: note: format string is defined here
  121 |             fprintf(file, "%-32s equ " SIZE_T_FMT "; %s\n", values[i].name, values[i].value, values[i].comment);
      |                                                      ~^
      |                                                       |
      |                                                       char *
      |                                                      %lld
/code/Vulkan-Loader/loader/asm_offset.c:121:27: warning: too many arguments for format [-Wformat-extra-args]
  121 |             fprintf(file, "%-32s equ " SIZE_T_FMT "; %s\n", values[i].name, values[i].value, values[i].comment);
      |                           ^~~~~~~~~~~~
/code/Vulkan-Loader/loader/asm_offset.c:137:27: warning: unknown conversion type character ‘z’ in format [-Wformat=]
  137 |             fprintf(file, ".set %-32s, " SIZE_T_FMT "%s %s\n", values[i].name, values[i].value, comment_delimiter,
      |                           ^~~~~~~~~~~~~~
/code/Vulkan-Loader/loader/asm_offset.c:62:24: note: format string is defined here
   62 | #define SIZE_T_FMT "%-8zu"
      |                        ^
/code/Vulkan-Loader/loader/asm_offset.c:137:27: warning: format ‘%s’ expects argument of type ‘char *’, but argument 4 has type ‘size_t’ {aka ‘long long unsigned int’} [-Wformat=]
  137 |             fprintf(file, ".set %-32s, " SIZE_T_FMT "%s %s\n", values[i].name, values[i].value, comment_delimiter,
      |                           ^~~~~~~~~~~~~~                                       ~~~~~~~~~~~~~~~
      |                                                                                         |
      |                                                                                         size_t {aka long long unsigned int}
/code/Vulkan-Loader/loader/asm_offset.c:137:55: note: format string is defined here
  137 |             fprintf(file, ".set %-32s, " SIZE_T_FMT "%s %s\n", values[i].name, values[i].value, comment_delimiter,
      |                                                      ~^
      |                                                       |
      |                                                       char *
      |                                                      %lld
/code/Vulkan-Loader/loader/asm_offset.c:137:27: warning: too many arguments for format [-Wformat-extra-args]
  137 |             fprintf(file, ".set %-32s, " SIZE_T_FMT "%s %s\n", values[i].name, values[i].value, comment_delimiter,
      |                           ^~~~~~~~~~~~~~
```


For more background:
Base msvcrt does not have a c99 compatible printf, so `%zu` is not actually supported. mingw-w64 does provide a printf replacement that has c99 support, but it requires setting `__USE_MINGW_ANSI_STDIO` for it to replace calls to `printf()` to its own.

I will add a comment regarding this macro and the usage of `%zu` for size_t

Reply:
```
#if !defined(_MSC_VER) || (_MSC_VER >= 1900)
#define SIZE_T_FMT "%-8zu"
#elif defined(__GNUC__) || defined(__clang__)
#define SIZE_T_FMT "%-8lu"
#else
#warning asm_offset.c SIZE_T_FMT must be defined for this platform
#endif
```
This shows that msvc (2015 and up) supports zu in printf, and the loader currently only supports VS 2017 and up. 

What is "Base msvcrt" in this context?

Reply:
Looking into `__USE_MINGW_ANSI_STDIO` a bit, it seems that its likely needed for mingw to behave correctly (like not print warnings or errors). So it seems like its fine to keep.